test/src/databases/all/functions.spec.ts

```diff
@@ -0,0 +1,1071 @@
+          databaseName === 'bigquery'
+            ? 'foo2003-01-01 12:00:00+00'
+            : 'foo2003-01-01 12:00:00',
+        ],
+        // TODO Maybe implement consistent null behavior
+        // ["concat('foo', null)", null],
+        ['concat()', '']
+      );
+    });
+  });
+
+  describe('round', () => {
+    it(`works - ${databaseName}`, async () => {
+      await funcTestMultiple(
+        ['round(1.2)', 1],
+        // TODO Remove when we upgrade to DuckDB 0.8.X -- DuckDB has some bugs with rounding
+        // that are fixed in 0.8.
+        ...(databaseName === 'duckdb_wasm'
+          ? []
+          : ([['round(12.222, 1)', 12.2]] as [string, number][])),
+        ['round(12.2, -1)', 10],
+        ['round(null)', null],
+        ['round(1, null)', null]
+      );
+    });
+  });
+
+  describe('floor', () => {
+    it(`works - ${databaseName}`, async () => {
+      await funcTestMultiple(
+        ['floor(1.9)', 1],
+        // TODO Remove when we upgrade to DuckDB 0.8.X -- DuckDB has some bugs with rounding
+        // that are fixed in 0.8.
+        ...(databaseName === 'duckdb_wasm'
+          ? []
+          : ([['floor(-1.9)', -2]] as [string, number][])),
+        ['floor(null)', null]
+      );
+      await funcTest('floor(1.9)', 1);
+    });
+  });
+
+  describe('ceil', () => {
+    it(`works - ${databaseName}`, async () => {
+      await funcTestMultiple(
+        ['ceil(1.9)', 2],
+        // TODO Remove when we upgrade to DuckDB 0.8.X -- DuckDB has some bugs with rounding
+        // that are fixed in 0.8.
+        ...(databaseName === 'duckdb_wasm'
+          ? []
+          : ([['ceil(-1.9)', -1]] as [string, number][])),
+        ['ceil(null)', null]
+      );
+    });
+  });
+
+  describe('length', () => {
+    it(`works - ${databaseName}`, async () => {
+      await funcTestMultiple(["length('foo')", 3], ['length(null)', null]);
+    });
+  });
+
+  describe('lower', () => {
+    it(`works - ${databaseName}`, async () => {
+      await funcTestMultiple(["lower('FoO')", 'foo'], ['lower(null)', null]);
+    });
+  });
+
+  describe('upper', () => {
+    it(`works - ${databaseName}`, async () => {
+      await funcTestMultiple(["upper('fOo')", 'FOO'], ['upper(null)', null]);
+    });
+  });
+
+  describe('regexp_extract', () => {
+    it(`works - ${databaseName}`, async () => {
+      await funcTestMultiple(
+        ["regexp_extract('I have a dog', r'd[aeiou]g')", 'dog'],
+        ["regexp_extract(null, r'd[aeiou]g')", null],
+        ["regexp_extract('foo', null)", null]
+      );
+    });
+  });
+
+  describe('replace', () => {
+    it(`works - ${databaseName}`, async () => {
+      await funcTestMultiple(
+        ["replace('aaaa', 'a', 'c')", 'cccc'],
+        ["replace('aaaa', r'.', 'c')", 'cccc'],
+        [
+          "replace('axbxc', r'(a).(b).(c)', '\\\\0 - \\\\1 - \\\\2 - \\\\3')",
+          databaseName === 'postgres' ? '\\0 - a - b - c' : 'axbxc - a - b - c',
+        ],
```

Comment:
DuckDB and BigQuery support using `\\0` to refer to "the whole match" when doing a regexp replace. Postgres doesn't.